### PR TITLE
Use to VectorScorer for exact vector scoring

### PR DESCRIPTION
**Pull request by benwtrent**
Lucene 9.11 introduced a new VectorScorer interface. We should utilize this interface when scoring exact vectors. 

related to: https://github.com/elastic/elasticsearch/issues/109293

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/search/vectors/ExactKnnQuery.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        private final byte[] query;
+
+        public Bytes(byte[] query, String field) {
+            super(field);
+            this.query = query;
+        }
+
+        @Override
+        public String toString(String field) {
+            return "ExactKnnQuery.Floats";
```
Comment (tteofili) [inline server/src/main/java/org/elasticsearch/search/vectors/ExactKnnQuery.java:140]: shouldn't be "ExactKnnQuery.Bytes"?

**Review by tteofili — APPROVED** [review]
LGTM, with a minor comment

**Review by ChrisHegarty — APPROVED** [review]
LGTM

**Review by pmpailis — APPROVED** [review]

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Pull request merged**